tiledb/sm/serialization/array_schema.cc

```diff
@@ -1190,7 +1196,8 @@ Status array_schema_serialize(
     const ArraySchema& array_schema,
     SerializationType serialize_type,
     SerializationBuffer& serialized_buffer,
-    const bool client_side) {
+    const bool client_side,
+    std::optional<std::string> storage_uri) {
```

Comment:
I can't comment in unchanged code, but we need to pass `storage_uri` to `array_schema_to_capnp` on 1206 below